### PR TITLE
CBZ 2: Tokyo Drift

### PR DESCRIPTION
**Pull request by WhyIsCaeciliusTaken**
## About The Pull Request

Readds CBZ and nerfs it to prevent people abusing it to mass check on ling rounds.

CBZ now uses a much harder chemical recipe (4u plasma, 1u N2O, 1u Anacea -> 1u CBZ) and cannot be ordered from cargo anymore.

CBZ no longer causes oxygen damage to lings, instead doing very minor stamina and brain damage, and making them scream. Non-lings are affected differently, taking massive, potentially fatal, brain damage. This prevents sec from injecting everyone in sight with the stuff to flush out changelings.

Fake CBZ also does this massive brain damage, but only to non-lings. Lings take minor brain damage to mask the fact that they are lings.

## Why It's Good For The Game

Speedmerging a poorly thought out removal and completely ignoring the consensus of the maintainers is dumb.

Encouraging sec players to abuse a code oversight to find changelings is dumb.

Spamming everyone who gets arrested with CBZ to find the lings is dumb.

## Changelog
:cl:
add: Added CBZ, again
add: CBZ and fake CBZ now do massive brain damage to non-lings, and minor brain damage to lings. Be very careful not to test an innocent person!
del: Removed CBZ cargo crate
balance: CBZ recipe now requires anacea
balance: CBZ now does not suffocate lings, making them scream instead.
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
> Speedmerging a poorly thought out removal and completely ignoring the consensus of the maintainers is dumb.

Giving cargo a practically-free, ranged, no-consequences ling detector in the first place was dumb.

**Comment by ike709** [comment]
> add: CBZ and fake CBZ now do massive brain damage to non-lings

I know the idea is to discourage blindly testing everyone, and it definitely should be discouraged, but this is just going to result in sec giving a bunch of people brain damage and we just have to hope that those players will receive treatment. Or worse, people using the chem specifically to brain damage people easily.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1904,6 +1904,56 @@
...
+	REMOVE_TRAIT(L, CHANGELING_HIVEMIND_MUTE, type)
+
+/datum/reagent/concentrated_bz/on_mob_life(mob/living/L)
+	if(L.mind)
+		var/datum/antagonist/changeling/changeling = L.mind.has_antag_datum(/datum/antagonist/changeling)
+		if(changeling)
+			if(prob(30))
+				L.adjustStaminaLoss(3,5)
+				L.emote("scream")
+				to_chat(L, "<font size=3 color=red><b>The chemicals burn our senses!</b></font>")
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1929]: Use a span class for all of the `to_chat()`s

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1904,6 +1904,56 @@
 			changeling.chem_charges = max(changeling.chem_charges-2, 0)
 	return ..()
 
+/datum/reagent/concentrated_bz
+	name = "Concentrated BZ"
+	description = "A hyperconcentrated liquid form of BZ gas, known to cause an extremely adverse reaction to changelings. Also causes minor brain damage."
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1909]: Mentions nothing of its effects on non-lings.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1904,6 +1904,56 @@
...
+				L.adjustOrganLoss(ORGAN_SLOT_BRAIN, 2, 50)
+		else
+			L.adjustOrganLoss(ORGAN_SLOT_BRAIN, 5, 200)
+			if(prob(30))
+				to_chat(L, "<font size=3 color=red><b>Your head feels like it's on fire!</b></font>")
+	..()
+
+/datum/reagent/fake_cbz
+	name = "Concentrated BZ"
+	description = "A hyperconcentrated liquid form of BZ gas, known to cause an extremely adverse reaction to changelings. Also causes minor brain damage."
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1939]: Mentions nothing of its effects on non-lings.

**Comment by utilbird** [comment]
Making CBZ less accessible does not address the underlying issue of being able to test lings wherever/whenever you want to. I mentioned this in the initial removal PR.
CBZ is a very unthoughtful way of near-instantly identifying antags. This discourages security from actually trying to find evidence and prove that someone is a ling, instead they're just going to inject the suspect with the magic reveal chemical.

A ling test in itself isn't a bad idea, but CBZ in both of these implementations isn't very interesting for security, chemists and especially the changeling. Making the recipe harder doesn't stop mass tests, it just means the chemist has to click a few more buttons for it to happen (this isn't even accounting for chem factories) 😅 

original comment: https://github.com/BeeStation/BeeStation-Hornet/pull/3353#issuecomment-757504591

**Comment by PowerfulBacon** [comment]
> I know the idea is to discourage blindly testing everyone, and it definitely should be discouraged, but this is just going to result in sec giving a bunch of people brain damage and we just have to hope that those players will receive treatment. Or worse, people using the chem specifically to brain damage people easily.

Sec brain damaging people is better than them just cremating them

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 3 above)
Comment (PowerfulBacon) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1939]: > Also causes minor brain damage.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1929]: \<span class='warning'> or \<span class='userdanger'>

**Comment by PowerfulBacon** [comment]
the effect should have a minimum tick before it takes place so you can't inject a couple of units and be done

**Comment by WhyIsCaeciliusTaken** [comment]
Chem factories are irrelevant as the recipe is deliberately impossible to make in chem factories. In the time it takes to make CBZ sec could make a much deadlier chem mix, or just take the lethal injection syringes and put them into a syringe gun.

Fentanyl + Neurotoxin is much easier and quicker to make than CBZ and is more dangerous to the brain.

**Comment by MCterra10** [comment]
steal goon's idea and require a chemical reaction with someone's blood to test for a ling

**Comment by eeSPee** [comment]
I liked an idea that came up on the forums, and I might implement myself.

The idea was that, instead of dealing damage to lings, a ling test should instead buff their abilities. This prevents testing people in public places (that is dangerous) or mass testing people. It will also give an incentive for lings to steal the ling tests and use them themselves. 

While I do not like the implementation of the idea that was proposed (the chemical would buff DNA regen), I would very much like it if it forced to mutate into a 'the thing' like abomination, with claw arms, a disfigured head, and greatly buffed combat capabilities (I'm thinking, complete stun immunity and a lot of damage to objects). This is a very high risk/reward mechanic.

On the negative side, it may end up with security downright ignoring this mechanic entirely, and relying on the current mechanics (debraining, or simply cremating a suspected ling) to mitigate the damage.

**Comment by utilbird** [comment]
Wew. That's a really good idea actually! As long as there's a small delay between the test and the mutation, so security doesn't immediately die. It'd have to give them enough time to inject the suspect in a cell and "run to cover"

**Comment by utilbird** [comment]
#3370

**Comment by Chessus** [comment]
> steal goon's idea and require a chemical reaction with someone's blood to test for a ling

I was thinking of this and heating it up, like an animation but code wise, does a sample of blood say that's X's blood and they have changeling trait ect?

**Comment by WhyIsCaeciliusTaken** [comment]
#3370 is a better and more fun solution than just nerfbatting the old version.

**Pull request closed without merging**